File: MyPython/MyPython/pi.py
import math
import random   #蒙特卡洛法求π
import time

points = 10**7      #总点数
hits = 0        #击中数
time.clock()
for i in range(1,points):
    x, y = random.random(), random.random()     #生成随机坐标
    radius = math.sqrt(x**2 + y**2)     #判断坐标与原点距离
    if radius <=1.0:        #判断命中
        hits = hits + 1
pi = 4 * (hits/points)
print(pi)
print(time.clock())

File: C/开方/开方/pi.c
#include<stdio.h>
#include<math.h>
#include<stdlib.h>
void main() {
	/*			蒙特卡洛法
	double r = 1.00, pi = 0.00;
	double x = 0.00, y = 0.00;
	double hits = 0, points = 1e+9;
	srand(time(NULL));
	for (int i = 0; i < points; i++)
	{
		x = (double)rand() / RAND_MAX;
		y = (double)rand() / RAND_MAX;
		if (sqrt(x*x + y*y) <= 1)
		{
			hits++;
		}
	}
	pi = (hits / points) * 4;
	printf("%lf\n", pi);
	*/
	//泰勒展开
	double k = 1.00, j = 0.00, sum = 1.00, pi = 0.00;
	for (int i = 1; i < 1e+9; i++)
	{
		if (i%2 == 0)
		{
			j = 1;
		}
		else
		{
			j = -1;
		}
		k = k + 2;
		sum += j*(1 / k);
	}
	pi = sum * 4;
	printf("%1.20lf\n", pi);
}